server/task/src/main/java/org/apache/james/task/TaskManagerWorker.java

```diff
@@ -0,0 +1,32 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.task;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import reactor.core.publisher.Mono;
+
+public interface TaskManagerWorker {
+
+    Mono<Task.Result> executeTask(TaskWithId taskWithId, Consumer<Function<TaskExecutionDetails, TaskExecutionDetails>> updateDetails);
```

Comment:
I can't understand this type: `Consumer<Function<TaskExecutionDetails, TaskExecutionDetails>> updateDetails`

Reply:
BTW I believe that the worker have to be responsible of updating task details.

Reply:
It's the worker wich update the task detail. Using the provided Function. I will try to simplify the type signature.

Reply:
Or maybe it should be another class responsibility: then you could do direct calls to update it?

Reply:
What do you think of the refactoring i have done using a functional interface ?

Reply:
I feel better ^^